hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/helpers/CloudObjectsSelector.java

```diff
@@ -0,0 +1,285 @@
+         ++i) {
+      List<Message> messages = sqsClient.receiveMessage(receiveMessageRequest).getMessages();
+      log.debug("Messages size: " + messages.size());
+
+      for (Message message : messages) {
+        log.debug("message id: " + message.getMessageId());
+        messagesToProcess.add(message);
+      }
+      log.debug("total fetched messages size: " + messagesToProcess.size());
+      if (messages.isEmpty() || (messagesToProcess.size() >= maxMessageEachBatch)) {
+        break;
+      }
+    }
+    return messagesToProcess;
+  }
+
+  /**
+   * create partitions of list using specific batch size. we can't use third party API for this
+   * functionality, due to https://github.com/apache/hudi/blob/master/style/checkstyle.xml#L270
+   */
+  protected List<List<Message>> createListPartitions(List<Message> singleList, int eachBatchSize) {
+    List<List<Message>> listPartitions = new ArrayList<>();
+
+    if (singleList.size() == 0 || eachBatchSize < 1) {
+      return listPartitions;
+    }
+
+    for (int start = 0; start < singleList.size(); start += eachBatchSize) {
+      int end = Math.min(start + eachBatchSize, singleList.size());
+
+      if (start > end) {
+        throw new IndexOutOfBoundsException(
+            "Index " + start + " is out of the list range <0," + (singleList.size() - 1) + ">");
+      }
+      listPartitions.add(new ArrayList<>(singleList.subList(start, end)));
+    }
+    return listPartitions;
+  }
+
+  /**
+   * delete batch of messages from queue.
+   */
+  protected void deleteBatchOfMessages(
+      AmazonSQS sqs, String queueUrl, List<Message> messagesToBeDeleted) {
+    DeleteMessageBatchRequest deleteBatchReq =
+        new DeleteMessageBatchRequest().withQueueUrl(queueUrl);
+    List<DeleteMessageBatchRequestEntry> deleteEntries = deleteBatchReq.getEntries();
+
+    for (Message message : messagesToBeDeleted) {
+      deleteEntries.add(
+          new DeleteMessageBatchRequestEntry()
+              .withId(message.getMessageId())
+              .withReceiptHandle(message.getReceiptHandle()));
+    }
+    DeleteMessageBatchResult deleteResult = sqs.deleteMessageBatch(deleteBatchReq);
+    List<String> deleteFailures =
+        deleteResult.getFailed().stream()
+            .map(BatchResultErrorEntry::getId)
+            .collect(Collectors.toList());
+    System.out.println("Delete is" + deleteFailures.isEmpty() + "or ignoring it.");
+    if (!deleteFailures.isEmpty()) {
+      log.warn(
+          "Failed to delete "
+              + deleteFailures.size()
+              + " messages out of "
+              + deleteEntries.size()
+              + " from queue.");
+    } else {
+      log.info("Successfully deleted " + deleteEntries.size() + " messages from queue.");
+    }
+  }
+
+  /**
+   * Delete Queue Messages after hudi commit. This method will be invoked by source.onCommit.
+   */
+  public void onCommitDeleteProcessedMessages(
```

Comment:
minor. rename to "deleteProcessedMessages". Every caller is already calling this from within onCommit(). so its understandable.